libraries/AP_Baro/AP_Baro_BMP085.cpp

```diff
@@ -237,5 +240,23 @@ bool AP_Baro_BMP085::_data_ready()
         return AP_HAL::millis() > _last_temp_read_command_time + 5;
     }
 
-    return AP_HAL::millis() > _last_press_read_command_time + 26;
+    uint32_t conversion_time_msec;
```

Comment:
OVERSAMPLING is known at compile time, which means the compiler will resolve the switch below at compile time.